posts/hello-world-machine.md

```diff
@@ -0,0 +1,252 @@
+
+```python
+def hello(greeting: str, location="World") -> str:
+    return greeting + ", " + location + "!"
+
+hello("Hello")
+```
+
+### The -join
+
+The machine already have some of the words, but it can't seen to string a ~~sen-~~ string together.
+
+```python
+def hello() -> str:
+    words = ["Hello", " world!"]
+    return ",".join(words)
+```
+
+#### The Title
+
+The machine's battery is a bit low today, all the words are lacking energy and not being capitalized properly.
+
+```python
+def hello() -> str:
+    words = "hello, world!"
+    return words.title()
+```
+
+#### The Reversal
+
+The machine somehow have the greeting ready, but something about it looks a bit ..backward.
+
+```python
+def hello() -> str:
+    sdrow = "!dlroW ,olleH"
+    return sdrow[::-1]
+```
+**Note:** `Reverse String` is a featured exercise for the #48in24 challenge, remember to check out various approaches of how to do it in different languages!
+
+#### The Slice 'n Dice
+
+Maybe the machine need to have its oil changed, the normal greeting message seem to be a bit mixed up.
+
+```python
+def hello() -> str:
+	w = "Holld, Werlo!"
+	return f"{w[0]}{w[8]}{w[2:4]}{w[1]}{w[5:8]}{w[-2]}{w[-4:-2]}{w[4]}{w[-1]}"
+```
+
+
+#### The Curry
+
+Oh oh, some parts of the machine is in maintenance mode. It forgot the greeting  message and now can only accept part by part back.
+
+```python
+def hello(greet: str):
+    def world(location: str) -> str:
+        return f"{greet}, {location}!"
+    return world
+
+hello = hello("Hello")
+hello("World")
+```
+
+#### The Ascii
+
+You accidentally spilled some lemonade on the machine and now it only show you these strange number instead of the usual characters in the message.
+
+```python
+def hello() -> str:
+    ascii = [72, 101, 108, 108, 111, 44, 32, 87, 111, 114, 108, 100, 33]
+    return ''.join(chr(value) for value in ascii)
+```
+
+#### The Dictionary
+
+You watched the news and saw a snippet about a man being rescued using S.O.S signal. "Maybe the machine should be able to handle morse code as well!" - you thought, but you are not sure why. Its only job is to return the greeting message...
+
+```python
+def hello(morse) -> str:
+    morse_code = {
+        "...."  : "H",
+        "."     : "e",
+        ".-.."  : "l",
+        "---"   : "o",
+        "--..--": ",",
+        "/"     : " ",
+        ".--"   : "W",
+        ".-."   : "r",
+        "-.."   : "d",
+        "-.-.--": "!"
+         
+    }
+    return "".join(morse_code[value] for value in morse.split())
+
+print(hello(".... . .-.. .-.. --- --..-- / .-- --- .-. .-.. -.. -.-.--"))
+```
+
+#### The Functional
+
+With your current machine are prone to changes and damage, you decide to build a smaller but an immutable and indestructible machine that do the exact same thing. And whenever the regular machine not working or acting up, it will activate the smaller one instead. Hang on, why don't we just use the smaller on from now on..
```

Comment:
```suggestion
With your current machine prone to changes and damage, you decide to build a smaller but immutable and indestructible machine that does the exact same thing.
Whenever the regular machine is not working or acting up, it will activate the smaller one instead.
Hang on, why don't we just use the smaller on from now on..
```